main.go

```diff
@@ -51,33 +52,70 @@ func init() {
 	//+kubebuilder:scaffold:scheme
 }
 
-func main() {
+type managerOpts struct {
```

Comment:
I really didn't like that all the options passed via args were just individual vars.

In use these vars did not indicate that they were input from the user at startup.

Now instead of cloudConfigFile we have opts.CloudConfigFile. This also helps to remove the opt vars from main, so for clarity alone, this is a win!